src/cascadia/TerminalControl/TSFInputControl.cpp

```diff
@@ -92,6 +93,27 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
         }
     }
 
+    // Method Description:
+    // - Clears the input buffer and tells the text server to clear their buffer as well.
+    //   Also clears the TextBlock and sets the active text starting point to 0.
+    // Arguments:
+    // - <none>
+    // Return Value:
+    // - <none>
+    void TSFInputControl::ClearBuffer()
+    {
+        if (!_inputBuffer.empty())
+        {
+            TextBlock().Text(L"");
+            _inputBuffer.clear();
+            _editContext.NotifyFocusLeave();
+            _editContext.NotifyTextChanged({ 0, ::base::ClampedNumeric<int32_t>(_inputBuffer.length()) }, 0, { 0, 0 });

```

Comment:
we're clearing the input buffer before we get its length; won't it be 0 here?

Reply:
oof you're totally right 😞

Reply:
Oof, didn't see this one before I hit approve. Good catch, @DHowett-MSFT .